### ISSUE TITLE
Detecting crashes from restored state

### ISSUE THREAD

**Issue by aphillipo** [pinqy520/mobx-persist#31]
Hi There,

Not sure if this have come up yet but if my app crashes within a shore amount of time of running hydrate I'd like to clear all the persistence.  Seems like a useful feature for obvious reasons, what do you thing?

**Comment by pinqy520** [comment]
React Native Android? Can you provide your crash logs?

**Comment by aphillipo** [comment]
Sorry to have not made myself clear: I want a feature that every time the App crashes within 200ms of a hydrate I clear that hydrate information. The reason I want this is if the App gets into a persistent crashing state there will be no way to recover. Does this make more sense as a feature request rather than an issue with your library?

**Comment by pinqy520** [comment]
OK, I understand now. You mean that some wrong persisted data will cause some JS errors, then this wrong data should be deleted. You can try this:

``` javascript
hydrate('some', someStore).catch(e => { 
    console.warn(e)
    // TODO: delete the state
})
```

**Comment by aphillipo** [comment]
Yes perfect start, just wondering if the catch will always fire or if there will be a small delay?
